src/Preprocessor.zig

```diff
@@ -689,6 +683,59 @@ fn expr(pp: *Preprocessor, tokenizer: *Tokenizer) MacroError!bool {
     return parser.macroExpr();
 }
 
+/// Turns macro_tok from .keyword_defined into .zero or .one depending on whether the argument is defined
+/// Returns the number of tokens consumed
+fn handleKeywordDefined(pp: *Preprocessor, macro_tok: *Token, tokens: []const Token, eof: RawToken) !usize {
+    std.debug.assert(macro_tok.id == .keyword_defined);
+    var it = TokenIterator.init(tokens);
+    const first = it.nextNoWS() orelse {
+        try pp.err(eof, .macro_name_missing);
+        return it.i;
+    };
+    switch (first.id) {
+        .identifier, .extended_identifier => {
```

Comment:
This has a bug similar to `__has_feature` that I fixed recently 7c179f6b9402b41e03bf60211bdecd183441d8f8:
```c
#define __const__ a

#if !defined __const__
#error isMacroIdentifier not used
#endif
```